src/play.jl

```diff
@@ -1,50 +1,132 @@
 """
-    GamePool
+    GamePool{N,S}
 
-A struct that defines a Wordle-like game.
+A struct that defines a Wordle-like game with targets of length `N`.
+`S <: Unsigned` is the smallest unsigned integer type that can store values in the
+range `0:(3 ^ N - 1)`, evaluated as `Wordlegames.scoretype(N)`
+
+The fields are:
+
+- `targetpool`: the target pool as a `Vector{NTuple{N,Char}}`
+- `guesspool`: a `Vector{NTuple{N,Char}}` of potential guesses (can be the same as `targets`)
+- `allscores`: a `Matrix{S}` where `allscores[i,j] = score(guesspool[j], targets[i])`
+- `active`: a `BitVector`. The active target pool is `targets[active]`.
+- `counts`: `Vector{Int}` of length `3 ^ N` in which bin counts are accumulated
+- `probs`: `Vector{Float64}` containing `counts ./ sum(counts)`
+- `guesses`: `Vector{NTuple{N,Char}}`
+- `guessinds`: `Vector{Int}` where the indexes into `guesspool` of guesses are stored
```

Comment:
likewise for all the other `Vector{Int}` pieces -- `S` will always be no larger than `Int` and will often be smaller. (If you even need to optimize that much for size.)